core/src/main/scala/cats/Representable.scala

```diff
@@ -0,0 +1,84 @@
+package cats
+
+/**
+ * Representable.
+ *
+ * Is a witness to the isomorphism forall A. F[A] <-> Representation => A
+ *
+ * Must obey the laws defined in cats.laws.RepresentableLaws
+ * i.e.
+ * tabulate andThen index = identity
+ * index andThen tabulate = identity
+ *
+ * Inspired by the Haskell representable package
+ * http://hackage.haskell.org/package/representable-functors-3.2.0.2/docs/Data-Functor-Representable.html
+ */
+trait Representable[F[_]] extends Serializable {
+
+  def F: Functor[F]
+
+  type Representation
+
+  /**
+   * Create a function that "indexes" into the `F` structure using `Representation`
+   */
+  def index[A](f: F[A]): Representation => A
+
+  /**
+   * Reconstructs the `F` structure using the index function
+   */
+  def tabulate[A](f: Representation => A): F[A]
+}
+
+private[cats] sealed abstract class RepresentableInstances0 extends RepresentableInstances1 {
+  def catsRepresentableIsBimonad[F[_], R](implicit Rep: Representable.Aux[F, R], Mon: Monoid[R]): Bimonad[F] = new RepresentableBimonad[F, R] {
+    override def R: Representable.Aux[F, R] = Rep
+    override def M: Monoid[R] = Mon
+  }
+}
+
+private[cats] sealed abstract class RepresentableInstances1 {
+  def catsRepresentableIsMonad[F[_], R](implicit Rep: Representable.Aux[F, R]): Monad[F] = new RepresentableMonad[F, R] {
```

Comment:
This and the one above are not implicit and not much point to be imported and thus no need to follow the implicits naming convention. If we want them to be available for auto derivation of `Bimonad` and `Monad` we need to put them in a trait extendable by `cats.implicits` and better have some tests for potential conflicts. I vote for just rename them to better searchable if there aren't many real world use cases for them.